spring-cloud-gcp-secretmanager/src/test/java/com/google/cloud/spring/secretmanager/SecretManagerPropertyUtilsTests.java

```diff
@@ -102,4 +102,109 @@ void testLongProperty_projectSecretVersion() {
     assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
     assertThat(secretIdentifier.getSecretVersion()).isEqualTo("3");
   }
+
+  @Test
+  void testNonRegionalSecret() {
+    String property = "spring.cloud.datasource";
+    SecretVersionName secretIdentifier =
+        SecretManagerPropertyUtils.getSecretVersionName(
+            property,
+            DEFAULT_PROJECT_ID_PROVIDER,
+            "us-central1"
+        );
+
+    assertThat(secretIdentifier).isNull();
+  }
+
+  @Test
+  void testInvalidSecretFormat_missingRegionalSecretId() {
+    String property = "sm://";
+
+    assertThatThrownBy(
+            () ->
+                    SecretManagerPropertyUtils.getSecretVersionName(
+                            property, DEFAULT_PROJECT_ID_PROVIDER, "us-central1"))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining("The GCP Secret Manager secret id must not be empty");
+  }
+
+  @Test
+  void testShortProperty_regionalSecretId() {
+    String property = "sm://the-reg-secret";
+    SecretVersionName secretIdentifier =
+        SecretManagerPropertyUtils.getSecretVersionName(
+            property,
+            DEFAULT_PROJECT_ID_PROVIDER,
+            "us-central1"
+        );
+
+    assertThat(secretIdentifier.getProject()).isEqualTo("defaultProject");
+    assertThat(secretIdentifier.getLocation()).isEqualTo("us-central1");
+    assertThat(secretIdentifier.getSecret()).isEqualTo("the-reg-secret");
+    assertThat(secretIdentifier.getSecretVersion()).isEqualTo("latest");
```

Comment:
In reference to my other comment, this is how AssertJ chaining performs several checks on the same object:

```java
assertThat(secretIdentifier)
  .returns("defaultProject", SecretVersionName::getProject)
  .returns("us-central1", SecretVersionName::getLocation)
  .returns("the-reg-secret", SecretVersionName::getSecret)
  .returns("latest", SecretVersionName::getSecretVersion);
```

Reply:
As mentioned in the previous comment, I opted to maintain consistency with other assertions used across the project.